### PR TITLE
Add tslint to the linting pipe

### PR DESCRIPTION
**Pull request by igor-dv**
Issue: Based on https://github.com/storybooks/storybook/pull/2679#discussion_r160078734

## What I did
Added tslint to the precommit + fixed linter errors.
Feel free to add new rules.

### CODE REVIEW

**Review by tmeasday — APPROVED** [review]
Seems reasonable to me!

File: package.json [hunk 1]
```diff
@@ -21,8 +21,9 @@
     "docs:deploy:manual": "npm --prefix docs run deploy:manual",
     "docs:dev": "npm --prefix docs run dev",
     "github-release": "github-release-from-changelog",
-    "lint": "yarn lint:js . && yarn lint:md .",
+    "lint": "yarn lint:js . && yarn lint:ts && yarn lint:md .",
     "lint:js": "cross-env NODE_ENV=production eslint --cache --cache-location=.cache/eslint --ext .js,.jsx,.json",
+    "lint:ts": "tslint -p . -t stylish -e '{**/node_modules/**,**/dist/**,lib/cli/test/**,**/storybook-static/**}' **/*.ts",
```
Comment (Hypnosphi) [inline package.json]: do they have something like `.tslintignore`?

Comment (igor-dv) [reply package.json]: Right, I'll check it!

**Comment by shilman** [comment]
adding to patch to reduce merge conflicts

**Pull request merged**